source/lib/login.ts

```diff
@@ -104,10 +103,10 @@ export function useCredentials<TData = QueryFnData, TError = DefaultError>(
 }
 
 export function useUsername(): QueryT<
-	false | Pick<SharedWebCredentials, 'username'>
+	null | Pick<SharedWebCredentials, 'username'>
 > {
 	return useCredentials({
-		select: (data) => (data ? {username: data.username} : false),
+		select: (data) => (data ? {username: data.username} : null),
```

Comment:
```suggestion
		select: (data) => data?.username,
```

Reply:
The types were unhappy when I tried this earlier

Reply:
What was the type error?

Reply:
I'll bet the explicit return type for `useUsername` has to adjust, too.

```
Type 'UseQueryResult<string | undefined, unknown>' is not assignable to type 'UseQueryResult<Pick<SharedWebCredentials, "username"> | null, unknown>'.
  Type 'QueryObserverRefetchErrorResult<string | undefined, unknown>' is not assignable to type 'UseQueryResult<Pick<SharedWebCredentials, "username"> | null, unknown>'.
    Type 'QueryObserverRefetchErrorResult<string | undefined, unknown>' is not assignable to type 'QueryObserverRefetchErrorResult<Pick<SharedWebCredentials, "username"> | null, unknown>'.
      Type 'string | undefined' is not assignable to type 'Pick<SharedWebCredentials, "username"> | null'.
        Type 'undefined' is not assignable to type 'Pick<SharedWebCredentials, "username"> | null'.ts(2322)
```

I think the difference here becomes

Before:
```
QueryT<null | Pick<SharedWebCredentials, 'username'>> 
```

After:
```
UseQueryResult<string | undefined, unknown>
```